docs: add repohasfile filter to search docs (#4573)
sourcegraph/sourcegraph: doc/user/search/queries.md

| **archived:no, archived:only**                                                    | Filter out results from archived repositories or filter results to only archived repositories. By default, results from archived repositories are included.                                                                                                                                                                                                                                                                                                                                                                                  | [`repo:sourcegraph/ archived:only`](https://sourcegraph.com/search?q=repo:%5Egithub.com/sourcegraph/+archived:only)                                                    |

> | **repohasfile:regexp-pattern** | Filter results to repositories that contain a matching file. This keyword is a pure filter, and requires at least one other search term in the query.  Note: this filter currently only works on text matches and file path matches. | [`repohasfile:\.py ubuntu`](https://sourcegraph.com/search?q=repogroup:sample+repohasfile:%5C.py+ubuntu)

Reviewer: ```suggestion
| **repohasfile:regexp-pattern** | Only show results from repositories that contain a matching file. This keyword is a pure filter, and requires at least one other search term in the query.  Note: this filter currently only works on text matches and file path matches. | [`repohasfile:\.py ubuntu`](https://sourcegraph.com/search?q=repogroup:sample+repohasfile:%5C.py+ubuntu)
```

> | **repohasfile:regexp-pattern** | Filter results to repositories that contain a matching file. This keyword is a pure filter, and requires at least one other search term in the query.  Note: this filter currently only works on text matches and file path matches. | [`repohasfile:\.py ubuntu`](https://sourcegraph.com/search?q=repogroup:sample+repohasfile:%5C.py+ubuntu)

Reviewer 2: ```suggestion
| **repohasfile:regexp-pattern** | Only include results from repositories that contain a matching file. As`repohasfile:regexp` is a filter, it requires one or more text or file path matches. | [`repohasfile:\.py file:Dockerfile repo:/sourcegraph/`](https://sourcegraph.com/search?q=repohasfile:%5C.py+file:Dockerfile+repo:/sourcegraph/)
```

Author: @ryan-blunden I like the change to not use "note:" but being a filter only means it requires another search term, not specifically a _text or file path_ search term. I am adding support for this to work with repo results, but it'll still require an additional search term (so it's still a filter), but it will no longer require specifically a file match or text match search term.

> | **-repohasfile:regexp-pattern** | Filter results to repositories that do not contain a matching file. This keyword is a pure filter, and requires at least one other search term in the query. Note: this filter currently only works on text matches and file path matches. | [`-repohasfile:Dockerfile docker`](https://sourcegraph.com/search?q=repogroup:sample+-repohasfile:Dockerfile+docker)

Reviewer: Use similar language as other negations: "Exclude results from repositories that contain a matching file."  
```suggestion
|
| **-repohasfile:regexp-pattern** | Exclude results from repositories that contain a matching file. This keyword is a pure filter, and requires at least one other search term in the query. Note: this filter currently only works on text matches and file path matches. | [`-repohasfile:Dockerfile docker`](https://sourcegraph.com/search?q=repogroup:sample+-repohasfile:Dockerfile+docker)
```
